### PR TITLE
Correctly refresh Java connections

### LINKED ISSUE #7003: Realm SyncManager takes up to 4-5 minutes to reconnect to ROS after waking device.
#### Goal

Keep realm fully synchronized at all times.

#### Actual Results
SyncManager does not automatically reconnect/refresh access tokens after raising device from sleep until 4 minutes have elapsed.

SyncManager refresh connections has no effect, does not immediately reconnect to ros, is affected by same delay.
SyncManager.stop() & SyncManager.start() has no effect. Even when called in onStart & onStop respectively.

Logging sync user out and back in resolves the issue immediately.

#### Steps & Code to Reproduce

Before you start:
1. Close realm in onDestroy
2. Disconnect device from power charging.

1. Login user
2. Open realm asynchronously.
3. Put device to sleep via power button with app in foreground.
4. Wait 4-5 minutes.
5. Wake device from sleep.
6. Realm will not reconnect to ros until 4-5 minutes later. Which means that any actions user makes within that time window are not synchronised to other devices via ros.

#### Version of Realm and tooling

Realm version(s): ?
6.0.2

Realm Sync feature enabled: Yes

Android Studio version: 4.0

Android Build Tools version: 28.0.3

Gradle version: ?

Which Android version and device(s): api level 19+ and not manufacturer specific.

### PR DESCRIPTION
**Pull request by cmelchior**
Closes #7003 

This fixes a bug where calling `SyncManager.refreshConnections()` didn't actually refresh connections that where in a backoff state on the Java side. This meant that even if you called `refreshConnections()` users could still end up waiting the maximum 5 minutes before the connection was established.

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/SyncSession.java [hunk 1]
```diff
@@ -888,26 +896,43 @@ protected void onSuccess(AuthenticateResponse response) {
...
+            @Override
+            protected String getName() {
+                return taskName;
             }
-        });
-        refreshTokenNetworkRequest = new RealmAsyncTaskImpl(task, SyncManager.NETWORK_POOL_EXECUTOR);
+        }, SyncManager.NETWORK_POOL_EXECUTOR);
+    }
+
+    void refreshConnection() {
```
Comment (rorbech) [inline realm/realm-library/src/objectServer/java/io/realm/SyncSession.java:919]: Shouldn't this be  synchronized on the SyncSession instance? Is it not reachable from user code through `SyncManager.refreshConnection`?

Comment (cmelchior) [reply realm/realm-library/src/objectServer/java/io/realm/SyncSession.java:919]: Yup. Good catch

File: realm/realm-library/src/objectServer/java/io/realm/SyncSession.java [hunk 2]
```diff
@@ -873,7 +881,7 @@ protected AuthenticateResponse execute() {
             protected void onSuccess(AuthenticateResponse response) {
                 synchronized (SyncSession.this) {
                     if (!isClosed && !Thread.currentThread().isInterrupted() && !refreshTokenNetworkRequest.isCancelled()) {
```
Comment (rorbech) [inline realm/realm-library/src/objectServer/java/io/realm/SyncSession.java:883]: Not sure that I have the full overview of all execution paths, but seems like `refreshTokenNetworkRequest` could have been cleared in the meantime, so should be checked for null?

**Comment by cmelchior** [comment]
@rorbech This unfortunately ballooned in size due to backporting the CI improvements and Plugin upgrades :(, so probably need another round of review... 

All the changes are just to accommodate the new CI setup and Findbugs though.

File: gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy [hunk 3]
```diff
@@ -121,7 +121,9 @@ class PluginTest {
     void pluginAddsRightRepositories_noRepositorySet() {
         project.buildscript {
             repositories {
-                google()
+                maven {
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy:124]: Yes, this is just a backport... All I remember from this was it was a big nightmare 🙈 😆

[the remark this answers is not in the document]
Comment (rorbech) [reply gradle-plugin/src/test/groovy/io/realm/gradle/PluginTest.groovy:124]: Yep, I stopped re-commenting when I stumbled upon the differently ordered repository lists.

File: realm/realm-library/src/objectServer/java/io/realm/SyncSession.java (hunk 2 above)
Comment (cmelchior) [reply realm/realm-library/src/objectServer/java/io/realm/SyncSession.java:883]: Looking into it. This should be covered by `isClosed` which is set before setting the `refreshTokenNetworkRequest` to null. So it should be safe.

**Pull request merged**